opensquirrel/utils/identity_filter.py

```diff
@@ -3,9 +3,19 @@
 from collections.abc import Iterable
 from typing import TYPE_CHECKING
 
+import numpy as np
+from numpy.typing import ArrayLike
+
+from opensquirrel.common import ATOL
+
 if TYPE_CHECKING:
     from opensquirrel.ir import Gate
 
 
 def filter_out_identities(gates: Iterable[Gate]) -> list[Gate]:
     return [gate for gate in gates if not gate.is_identity()]
+
+
+def is_matrix_identity(matrix: ArrayLike) -> bool:
+    norm_matrix = np.abs(matrix)
+    return np.allclose(norm_matrix, np.eye(norm_matrix.shape[0]), atol=ATOL)
```

Comment:
Yes we need to do the abs because the matrix can be a complex number. This is what was going wrong before with the phase. If you change the phase, you introduce a global phase of e^(phi * i). When the `is_identity` function would be called, it would return `False`. This extra step makes sure that the global phase is accounted for, whether it be negative or complex.

Reply:
OK, can we add a parametrized test for `is_identity_matrix` with a good deal of cases?

My point is that `is_identity_matrix([[-1, 0], [0, 1]])` is returning true but it shouldn't should it?